api_app/script_analyzers/observable_analyzers/dnsdb.py

```diff
@@ -33,20 +121,133 @@ def run(self):
         if self.observable_classification == "ip":
             endpoint = "rdata/ip"
         elif self.observable_classification in ["domain", "url"]:
-            endpoint = "rrset/name"
+            if self.query_type == "domain":
+                endpoint = "rrset/name"
+            elif self.query_type == "rrname-wildcard-left":
+                endpoint = "rrset/name"
+                observable_to_check = "*." + observable_to_check
+            elif self.query_type == "rrname-wildcard-right":
+                endpoint = "rrset/name"
+                observable_to_check += ".*"
+            elif self.query_type == "names":
+                endpoint = "rdata/name"
+            elif self.query_type == "rdata-wildcard-left":
+                endpoint = "rdata/name"
+                observable_to_check = "*." + observable_to_check
+            elif self.query_type == "rdata-wildcard-right":
+                endpoint = "rdata/name"
+                observable_to_check += observable_to_check + ".*"
+            else:
+                raise AnalyzerRunException(f"{self.query_type} not supported")
         else:
             raise AnalyzerRunException(
                 f"{self.observable_classification} not supported"
             )
 
-        url = f"https://api.dnsdb.info/lookup/{endpoint}/{observable_to_check}"
+        url = (
+            f"https://{self.dnsdb_server}{api_version}/lookup/{endpoint}"
+            f"/{observable_to_check}/{self.rrtype}"
+        )
         params = {"limit": self.limit}
         response = requests.get(url, params=params, headers=headers)
         response.raise_for_status()
         results_list = response.text
+
+        # different versions have different parsers
         json_extracted_results = []
-        for item in results_list.split("\n"):
-            if item:
-                json_extracted_results.append(json.loads(item))
+        if self.api_version == 2:
+            # first elem is a context line, last two are a context line and a empty line
+            for item in results_list.split("\n")[1:-2]:
+                if item:
+                    new_element = json.loads(item)
+                    # response element is wrapped in object field
+                    json_extracted_results.append(new_element["obj"])
+        elif self.api_version == 1:
+            for item in results_list.split("\n"):
+                if item:
+                    json_extracted_results.append(json.loads(item))
+        else:
+            raise AnalyzerRunException(
+                f"{self.api_version} not supported version, "
+                f"available versions: {supported_version}"
+            )
+
+        # filter results for max count value
```

Comment:
it should be possible to perform this filters at the time of the query. We should check the docs again